HoaryFox/Member/CreateTag.cs

```diff
@@ -14,134 +15,167 @@ namespace HoaryFox.Member
-                                break;
-                            case FrameType.Slab:
-                                break;
-                            case FrameType.Wall:
-                                break;
-                            case FrameType.Any:
-                                break;
-                            default:
-                                throw new ArgumentOutOfRangeException();
-                        }
-                        shapeType = p1 <= 0 ? ShapeTypes.Pipe : ShapeTypes.BOX;
+                        tagInfo = TagRc(frame, idSection);
                         break;
                     case KindsStructure.S:
-                    {
-                        int idShape;
-                        switch (frame.FrameType)
-                        {
-                            case FrameType.Column:
-                            case FrameType.Post:
-                                idShape = colS.Id.IndexOf(idSection);
-                                shape = colS.Shape[idShape];
-                                break;
-                            case FrameType.Girder:
-                            case FrameType.Beam:
-                                idShape = beamS.Id.IndexOf(idSection);
-                                shape = beamS.Shape[idShape];
-                                break;
-                            case FrameType.Brace:
-                                idShape = braceS.Id.IndexOf(idSection);
-                                shape = braceS.Shape[idShape];
-                                break;
-                            case FrameType.Slab:
-                                break;
-                            case FrameType.Wall:
-                                break;
-                            case FrameType.Any:
-                                break;
-                            default:
-                                throw new ArgumentOutOfRangeException();
-                        }
-        
-                        secIndex = secSteel.Name.IndexOf(shape);
-                        name = secSteel.Name[secIndex];
-                        p1 = secSteel.P1[secIndex];
-                        p2 = secSteel.P2[secIndex];
-                        p3 = secSteel.P3[secIndex];
-                        p4 = secSteel.P4[secIndex];
-                        shapeType = secSteel.ShapeType[secIndex];
+                        tagInfo = TagSteel(frame, idSection);
                         break;
-                    }
                     case KindsStructure.Src:
-                        break;
                     case KindsStructure.Cft:
-                        break;
                     case KindsStructure.Deck:
-                        break;
                     case KindsStructure.Precast:
-                        break;
                     case KindsStructure.Other:
-                        break;
+                        throw new Exception("Wrong kind structure");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/hrntsm/HoaryFox/pullRequest?prid=6419994)

Reply:
fix